src/domain/dataset-update-flow/src/entities/update/update_delay_reason.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::Duration;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum UpdateDelayReason {
+    Throttling(UpdateThrottlingDelayReason),
+    Batching(UpdateBatchingDelayReason),
+}
```

Comment:
Not sure about this.

Would it be better to model this not as a delay reason but as an `UpdateStartCondition`?

i.e. dataset updates system continuously evaluates new incoming **triggers** and based on them decides what is the **start condition** of the next update (immediate, throttled, pending batch).

Reply:
I originally saw this as a way to cover provenance history needs. Perhaps, modeling it as a state as well will be needed to implement the evaluations - this part is still to be defined, so I wan't not in a hurry to define it as state.